Sources/StreamChatUI/ChatMessageList/ChatMessageListCollectionView.swift

```diff
@@ -306,3 +306,252 @@ open class ChatMessageListCollectionView<ExtraData: ExtraDataTypes>: UICollectio
+        let scrollOverlayView = components.messageListScrollOverlayView.init()
+        scrollOverlayView.isHidden = true
+        return scrollOverlayView.withoutAutoresizingMaskConstraints
+    }()
+
+    override open func didMoveToSuperview() {
+        super.didMoveToSuperview()
+        
+        guard !isInitialized, superview != nil else { return }
+
+        isInitialized = true
+
+        setUp()
+        setUpLayout()
+        setUpAppearance()
+    }
+    
+    open func setUp() {
+        keyboardDismissMode = .onDrag
+        rowHeight = UITableView.automaticDimension
+        estimatedRowHeight = 150
+        separatorStyle = .none
+        transform = .mirrorY
+        scrollOverlayView.transform = .mirrorY
+            
+        // Setup `contentOffset` observation so `delegate` is free for anyone that wants to use it
+        contentOffsetObservation = observe(\.contentOffset) { tb, _ in
+            /// To display correct date we use bottom edge of `dateView` (we use `cv.layoutMargins.top` for both vertical offsets of `dateView`
+            let dateViewRefPoint = CGPoint(
+                x: tb.scrollOverlayView.center.x,
+                y: tb.scrollOverlayView.frame.minY
+            )
+            
+            // If we cannot find any indexPath for `cell` we try to use max visible indexPath (we have bottom to top) layout
+            guard let indexPath = tb.indexPathForRow(at: dateViewRefPoint) ?? tb.indexPathsForVisibleRows?.max() else { return }
+            
+            let overlayText = tb.chatDataSource?.tableView(tb, scrollOverlayTextForItemAt: indexPath)
+            
+            // As cells can overlay our `dateView` we need to keep it above them
+            tb.bringSubviewToFront(tb.scrollOverlayView)
+            
+            // If we have no date we have no reason to display `dateView`
+            tb.scrollOverlayView.isHidden = (overlayText ?? "").isEmpty
+            tb.scrollOverlayView.content = overlayText
+            
+            // Apple's naming is quite weird as actually this property should rather be named `isScrolling`
+            // as it stays true when user stops dragging and scrollView is decelerating and becomes false
+            // when scrollView stops decelerating
+            //
+            // But this case doesn't cover situation when user drags scrollView to a certain `contentOffset`
+            // leaves the finger there for a while and then just lifts it, it doesn't change `contentOffset`
+            // so this handler is not called, this is handled by `scrollStateChanged`
+            // that reacts on `panGestureRecognizer` states and can handle this case properly
+            if !tb.isDragging {
+                tb.setOverlayViewAlpha(0)
+            }
+        }
+        
+        panGestureRecognizer.addTarget(self, action: #selector(scrollStateChanged))
+    }
+    
+    open func setUpAppearance() { /* default empty implementation */ }
+    open func setUpLayout() {
```

Comment:
call `super` if you added default content to those methods.

Reply:
that's the `UITableView` subclass so there's no super implementation. Do you think we need a base `_TableView`? 🤔 Message list it's the only place where we use a table view so far

Reply:
I think I saw it's a subclass of one of our views. My bad, sorry!